src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Core.cs

```diff
@@ -62,5 +64,33 @@ private void CheckDisposedOrStarted()
                 throw new InvalidOperationException(SR.net_http_operation_started);
             }
         }
+
+        private void ThrowForModifiedManagedSslOptionsIfStarted()
+        {
+            // Hack to trigger an InvalidOperationException if a property that's stored on
+            // SslOptions is changed, since SslOptions itself do any such checks.
+            _managedHandler.SslOptions = _managedHandler.SslOptions;
+        }
+
+        /// <summary>Helper type used to map a ServerCertificateCustomValidationCallback to a RemoteCertificateValidationCallback.</summary>
+        private sealed class ServerCertificateValidationCallbackManagedMapper
+        {
+            public Func<HttpRequestMessage, X509Certificate2, X509Chain, SslPolicyErrors, bool> ServerCertificateCustomValidationCallback { get; set; }
```

Comment:
Nit: could this be passed to the constructor instead of a mutable property?

Reply:
With it being mutable it means that setting the property twice doesn't require allocating a new object, it just reuses the same one.  But I can simplify it... not something that really needs to be optimized for.